bmc/firmware.go

```diff
@@ -16,7 +16,7 @@ type BMCVersionGetter interface {
 
 // BMCFirmwareUpdater upgrades the BMC firmware
 type BMCFirmwareUpdater interface {
-	FirmwareUpdateBMC(ctx context.Context, fileReader io.Reader) (err error)
+	FirmwareUpdateBMC(ctx context.Context, fileReader io.Reader, fileSize int64) (err error)
```

Comment:
if this generic enough for hardware that isn't ASRR?

Reply:
Im not too happy with the func signature there, and it is subject to change - In another iteration, I'd prefer to just pass in the URL to the file and let the method wrap the rest.

For now this will work with Redfish on the Dells and SMC - since they both use multi-part form upload.

HPE requires a URL https://github.com/HewlettPackard/python-redfish-utility/blob/master/examples/Linux/Update_Firmware.sh#L38